Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,225 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TSCBasic
+import Splash
+
+private extension StringProtocol {
+  func matches(regex: NSRegularExpression) -> String.SubSequence? {
+    let str = String(self)
+    guard let range = str.range(of: regex),
+          range.upperBound < str.endIndex
+          else { return nil }
+    return str[range.upperBound..<str.endIndex]
+  }
+
+  func range(of regex: NSRegularExpression) -> Range<String.Index>? {
+    let str = String(self)
+    let range = NSRange(location: 0, length: utf16.count)
+    guard let match = regex.firstMatch(in: str, options: [], range: range),
+          let matchRange = Range(match.range, in: str) else {
+      return nil
+    }
+    return matchRange
+  }
+}
+
+private extension String.StringInterpolation {
+  mutating func appendInterpolation<T>(_ value: T, color: String...) {
+    appendInterpolation("\(color.map { "\u{001B}\($0)" }.joined())\(value)\u{001B}[0m")
+  }
+}
+
+private extension TokenType {
+  var color: String {
+    // Reference on escape codes: https://en.wikipedia.org/wiki/ANSI_escape_code#Colors
+    switch self {
+    case .keyword: return "[35;1m" // magenta;bold
+    case .comment: return "[90m" // bright black
+    case .call, .dotAccess, .property, .type: return "[94m" // bright blue
+    case .number, .preprocessing: return "[33m" // yellow
+    case .string: return "[91;1m" // bright red;bold
+    default: return "[0m" // reset
+    }
+  }
+}
+
+private struct TerminalOutputFormat: OutputFormat {
+  func makeBuilder() -> TerminalOutputBuilder {
+    .init()
+  }
+
+  struct TerminalOutputBuilder: OutputBuilder {
+    var output: String = ""
+
+    mutating func addToken(_ token: String, ofType type: TokenType) {
+      output.append("\(token, color: type.color)")
+    }
+
+    mutating func addPlainText(_ text: String) {
+      output.append(text)
+    }
+
+    mutating func addWhitespace(_ whitespace: String) {
+      output.append(whitespace)
+    }
+
+    mutating func build() -> String {
+      output
+    }
+  }
+}
+
+/// Parses and re-formats diagnostics output by the Swift compiler.
+///
+/// The compiler output often repeats iteself, and the diagnostics can sometimes be difficult to read.
+/// This reformats them to a more readable output.
+struct DiagnosticsParser {
+  enum Regex {
+    /// The output has moved to a new file
+    static let enterFile = try! NSRegularExpression(pattern: #"\[\d+\/\d+\] Compiling \w+ "#)
```

Comment:
<!--
  1 failure:  Force tries shoul...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: Sources/CartonHelpers/DiagnosticsParser.swift;
  Line: 92;
-->
- :no_entry_sign: Force tries should be avoided. (`force_try`)